src/it/java/org/checkstyle/plugins/sonar/RunPluginTest.java

```diff
@@ -0,0 +1,163 @@
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import com.sonar.orchestrator.Orchestrator;
+import com.sonar.orchestrator.OrchestratorBuilder;
+import com.sonar.orchestrator.build.Build;
+import com.sonar.orchestrator.build.BuildResult;
+import com.sonar.orchestrator.build.MavenBuild;
+import com.sonar.orchestrator.build.SonarScanner;
+import com.sonar.orchestrator.locator.FileLocation;
+import com.sonar.orchestrator.locator.Location;
+import com.sonar.orchestrator.locator.MavenLocation;
+import com.sonar.orchestrator.locator.URLLocation;
+import org.sonarqube.ws.WsMeasures;
+import org.sonarqube.ws.client.HttpConnector;
+import org.sonarqube.ws.client.WsClient;
+import org.sonarqube.ws.client.WsClientFactories;
+import org.sonarqube.ws.client.measure.ComponentWsRequest;
+
+/**
+ * a lot of code from https://github.com/SonarSource/sonar-java/blob/5aaaeba90cd85a40391706ad2f806d6a156f80cb/its/ruling/src/test/java/org/sonar/java/it/JavaRulingTest.java
+ * and from
+ * https://github.com/SonarSource/sonar-jacoco/blob/master/its/src/test/java/org/sonar/plugins/jacoco/its/JacocoTest.java
+ */
+class RunPluginTest {
+
+    private final static String PROJECT_KEY = "jacoco-test-project";
+  private static final String FILE_KEY = "jacoco-test-project:src/main/java/org/sonarsource/test/Calc.java";
+  private static final String FILE_WITHOUT_COVERAGE_KEY = "jacoco-test-project:src/main/java/org/sonarsource/test/CalcNoCoverage.java";
+
+
+    private static Orchestrator orchestrator;
+
+    @BeforeClass
+    public static void beforeAll() {
+        Orchestrator orchestrator = Orchestrator.builderEnv()
+          .setSonarVersion(System.getProperty("sonar.runtimeVersion", "LATEST_RELEASE[7.9]"))
+          .addPlugin(FileLocation.byWildcardMavenFilename(new File("../../sonar-java-plugin/target"), "sonar-java-plugin-*.jar"))
+          //.addPlugin(MavenLocation.of("org.sonarsource.java", "sonar-java-plugin", "5.2.0.13398")
+          .setServerProperty("sonar.web.javaOpts", "-Xmx1G")
+          .build();
+
+        // start server
+        orchestrator.start();
+    }
+
+    @AfterClass
+    public static void afterAll() {
+        orchestrator.stop();
+    }
+
+    @Test
+    public void run() {
+        SonarScanner build = SonarScanner.create()
+            .setProjectKey(PROJECT_KEY)
+            .setDebugLogs(true)
+            .setSourceDirs("src/main")
+            .setTestDirs("src/test")
+            .setProperty("sonar.java.binaries", ".")
+            .setProjectDir(prepareProject("simple-project-jacoco"));
+        orchestrator.executeBuild(build);
+
+    MavenBuild build = test_project("com.puppycrows.tools:checkstyle", "src/");
+    executeBuildWithCommonProperties(build, projectName, false);
+
+        List<String> metricKeys = Arrays.asList("line_coverage", "lines_to_cover",
```

Comment:
please clear code a bit with indentation ...
It is kind of strange why checksyle validation does not detect this ....

Reply:
chekcstyle is updated - https://github.com/checkstyle/sonar-checkstyle/pull/260

Reply:
@romani Checkstyle is not running on these files since they are in `it` folder. Checkstyle maven run has to be updated with this new folder.

Reply:
thanks a lot !! 
I finally found root reason - it is resolved at #260 .

Reply:
@romani It is not resolved as CI still allows the same code in.

Reply:
will be addressed at #269